# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmTypes.cpp
# Commit: 5737845e8af2
# Full Hash: 5737845e8af228fab6e5e622863663faa50f93e8
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-30 16:09:44
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth
#   
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
# ==============================================================================

diff -r 5dfb9fc15fba -r 5737845e8af2 js/src/wasm/WasmTypes.cpp
--- a/js/src/wasm/WasmTypes.cpp	Fri Oct 23 18:17:37 2020 +0000
+++ b/js/src/wasm/WasmTypes.cpp	Fri Oct 23 18:17:38 2020 +0000
@@ -602,10 +602,11 @@
 
 /* static */
 DebugFrame* DebugFrame::from(Frame* fp) {
-  MOZ_ASSERT(fp->instance()->code().metadata().debugEnabled);
+  MOZ_ASSERT(
+      GetNearestEffectiveTls(fp)->instance->code().metadata().debugEnabled);
   auto* df =
       reinterpret_cast<DebugFrame*>((uint8_t*)fp - DebugFrame::offsetOfFrame());
-  MOZ_ASSERT(fp->instance() == df->instance());
+  MOZ_ASSERT(GetNearestEffectiveTls(fp)->instance == df->instance());
   return df;
 }
 
@@ -626,6 +627,10 @@
 #endif
 }
 
+Instance* DebugFrame::instance() const {
+  return GetNearestEffectiveTls(&frame_)->instance;
+}
+
 GlobalObject* DebugFrame::global() const {
   return &instance()->object()->global();
 }